### PR TITLE
Update to CircleCI 2.0 config

### PR DESCRIPTION
**Pull request by jerryblakley**

### CODE REVIEW

**Comment by jerryblakley** [comment]
https://getpantheon.atlassian.net/browse/PLAT-799

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+         command: |
+           set +eo pipefail
+           /usr/local/bin/cpplint --filter=-whitespace/line_length src/*
+           true
+      - run:
+         name: Build the image
+         command: bash scripts/docker-outer.sh
+      - *persist_to_workspace
+
+  deploy-dev:
```
Comment (winmillwill) [inline .circleci/config.yml]: You may want to use our deploy-toolbox image. Not sure if the fusedav handling of packagecloud is and should be different from normal.

Comment (winmillwill) [reply .circleci/config.yml]: all seems fine otherwise if the only thing to fix is the volume deletion error.

**Comment by winmillwill** [comment]
All seems fine, pending a fix for the volume deletion.

👍

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
+---
+# all-purpose workspace persister
+persist_to_workspace: &persist_to_workspace
```
Comment (spheromak) [inline .circleci/config.yml]: can use commands with teh 2.1 config to achieve this. 

example here: https://github.com/pantheon-systems/gke-terraform/blob/master/.circleci/config.yml#L23

Personally I like the commands vs the yaml macro.

**Pull request merged**